plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java

```diff
@@ -94,46 +95,51 @@ private static Session getTestSession(Identity identity)
     @Test
     public void testCacheRefreshOnGrantAndRevoke()
     {
-        assertThatThrownBy(() -> queryRunner.execute(ALICE, "SELECT * FROM test"))
+        assertThatThrownBy(() -> getQueryRunner().execute(ALICE, "SELECT * FROM test"))
                 .hasMessageContaining("Access Denied");
-        queryRunner.execute("GRANT SELECT ON test TO " + ALICE_NAME);
-        queryRunner.execute(ALICE, "SELECT * FROM test");
-        queryRunner.execute("REVOKE SELECT ON test FROM " + ALICE_NAME);
-        assertThatThrownBy(() -> queryRunner.execute(ALICE, "SELECT * FROM test"))
+        getQueryRunner().execute("GRANT SELECT ON test TO " + ALICE_NAME);
+        getQueryRunner().execute(ALICE, "SELECT * FROM test");
+        getQueryRunner().execute("REVOKE SELECT ON test FROM " + ALICE_NAME);
+        assertThatThrownBy(() -> getQueryRunner().execute(ALICE, "SELECT * FROM test"))
                 .hasMessageContaining("Access Denied");
     }
 
     @Test(dataProvider = "testCacheRefreshOnRoleGrantAndRevokeParams")
     public void testCacheRefreshOnRoleGrantAndRevoke(List<String> grantRoleStatements, String revokeRoleStatement)
     {
-        assertThatThrownBy(() -> queryRunner.execute(ALICE, "SELECT * FROM test"))
+        assertThatThrownBy(() -> getQueryRunner().execute(ALICE, "SELECT * FROM test"))
                 .hasMessageContaining("Access Denied");
+        QueryRunner queryRunner = getQueryRunner();
```

Comment:
Why is this assignment needed? We are not going to get a "new" query runner.

Reply:
Changed. Now we always get new query runner.